client/my-sites/checkout/composite-checkout/composite-checkout.tsx

```diff
@@ -550,6 +557,56 @@ export default function CompositeCheckout( {
 		reduxDispatch( infoNotice( translate( 'Redirecting to payment partner…' ) ) );
 	}, [ reduxDispatch, translate ] );
 
+	// The goToPreviousPage function and subsequent conditional statement controls the 'back' button functionality on the empty cart page
+
+	const checkoutBackUrl = useValidCheckoutBackUrl( siteSlug );
+
+	const goToPreviousPage = useCallback( () => {
+		let closeUrl = siteSlug ? '/plans/' + siteSlug : '/start';
+
+		reduxDispatch( recordTracksEvent( 'EMPTY_CART_CTA_CLICKED' ) );
+
+		if ( checkoutBackUrl ) {
+			window.location.href = checkoutBackUrl;
+			return;
+		}
+
+		if (
+			previousPath &&
+			'' !== previousPath &&
+			previousPath !== window.location.href &&
+			! previousPath.includes( '/checkout/' )
+		) {
+			closeUrl = previousPath;
+		}
+
+		try {
+			const searchParams = new URLSearchParams( window.location.search );
+
+			if ( searchParams.has( 'signup' ) ) {
+				clearSignupDestinationCookie();
+			}
+
+			// Some places that open checkout (eg: purchase page renewals) return the
+			// user there after checkout by putting the previous page's path in the
+			// `redirect_to` query param. When leaving checkout via the close button,
+			// we probably want to return to that location also.
+			if ( searchParams.has( 'redirect_to' ) ) {
+				const redirectPath = searchParams.get( 'redirect_to' ) ?? '';
+				// Only allow redirecting to relative paths.
+				if ( redirectPath.startsWith( '/' ) ) {
+					page( redirectPath );
+					return;
+				}
+			}
+		} catch ( error ) {
+			// Silently ignore query string errors (eg: which may occur in IE since it doesn't support URLSearchParams).
+			console.error( 'Error getting query string in close button' ); // eslint-disable-line no-console
+		}
+
+		window.location.href = closeUrl;
```

Comment:
I think we could probably use `page()` here too, although we should verify that it's still a relative URL like we do above.

```js
				if ( closeUrl.startsWith( '/' ) ) {
					page( closeUrl );
					return;
				}
               window.location.href = closeUrl;
```